ElrondNetwork/elrond-go
# EN-13311: fix delegation contract's owner delegation logs

## Reasoning behind the pull request
- The `_init` function from delegation logs didn't have the first argument as the delegator address, so the `Address` field of the `delegate` event was not filled with the owner's address
- Added logs for `changeOwner` function 
  
## Proposed Changes
- Don't fetch the address from logs if the function is `_init`, but rather get it from the contract call input

## Testing procedure
- create a new delegation contract: https://docs.elrond.com/validators/delegation-manager/#creating-a-new-delegation-contract
- in the logs & events tab, you should see the owner's address as the `Address` field in the `delegate` log, as opposed to this:
https://explorer.elrond.com/transactions/24ab9b142606002bfb731d40e1c5265b18a164caee2680d421d6e17020866b21/logs
